packages/axelar-soroban-std/src/interfaces/operatable.rs

```diff
@@ -0,0 +1,145 @@
+            Symbol::new(env, "operatorship_transferred"),
+            self.previous_operator.to_val(),
+            self.new_operator.to_val(),
+        )
+    }
+
+    fn data(&self, env: &Env) -> impl IntoVal<Env, Val> + Debug {
+        Vec::<Val>::new(env)
+    }
+}
+
+#[cfg(any(test, feature = "testutils"))]
+impl_event_testutils!(OperatorChangedEvent, (Symbol, Address, Address), ());
+
+// submodule to encapsulate the disabled linting
+mod storage {
+    // linting is disabled for the enum variant names on purpose, so we can define names that would otherwise be invalid.
+    // This way, if a contract that implements a shared interface defines a variant with the same name, the linter will
+    // complain about it.
+    #![allow(non_camel_case_types)]
+
+    use soroban_sdk::contracttype;
+
+    #[contracttype]
+    /// Variants do not follow the naming convention of other variants to let the linter help to avoid
+    /// collisions with contract types defined in other contracts that implement a shared interface.
+    pub enum DataKey {
+        Interfaces_Operator,
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::interfaces::testdata::contract::Contract;
+    use crate::interfaces::{OperatableClient, OperatorChangedEvent};
+    use crate::{assert_invoke_auth_err, assert_invoke_auth_ok, events};
+    use soroban_sdk::testutils::Address as _;
+    use soroban_sdk::{Address, Env};
+
+    #[test]
+    fn operator_fails_if_operator_not_set() {
+        let env = Env::default();
+        let client = prepare_client(&env, None);
+
+        assert!(client.try_operator().is_err());
+    }
+
+    #[test]
+    fn operator_returns_correct_operator_when_set() {
+        let env = Env::default();
+        let operator = Address::generate(&env);
+        let client = prepare_client(&env, Some(operator.clone()));
+
+        assert_eq!(client.operator(), operator);
+    }
+
+    #[test]
+    fn transfer_operatorship_fails_if_caller_is_not_operator() {
+        let env = Env::default();
+        let operator = Address::generate(&env);
+        let client = prepare_client(&env, Some(operator));
+
+        let new_operator = Address::generate(&env);
+        assert_invoke_auth_err!(
+            new_operator,
+            client.try_transfer_operatorship(&new_operator)
+        );
+    }
+
+    #[test]
+    fn transfer_operatorship_succeeds_if_caller_is_operator() {
+        let env = Env::default();
+        let operator = Address::generate(&env);
+        let client = prepare_client(&env, Some(operator.clone()));
+
+        assert_eq!(client.operator(), operator);
+
+        let new_operator = Address::generate(&env);
+        assert_invoke_auth_ok!(operator, client.try_transfer_operatorship(&new_operator));
+
+        goldie::assert!(events::fmt_last_emitted_event::<OperatorChangedEvent>(&env));
+
+        assert_eq!(client.operator(), new_operator);
+    }
+
+    fn prepare_client(env: &Env, operator: Option<Address>) -> OperatableClient {
```

Comment:
nit: I think we have a convention that setting up or preparing the environment should be at the top of the unit test code. Could you move this to around line 93?